SchedulearnBackend/SchedulearnBackend/Services/UserService.cs

```diff
@@ -1,27 +1,61 @@
-﻿using Microsoft.EntityFrameworkCore;
+﻿using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.IdentityModel.Tokens;
 using SchedulearnBackend.Controllers.DTOs;
 using SchedulearnBackend.DataAccessLayer;
 using SchedulearnBackend.Models;
 using SchedulearnBackend.UserFriendlyExceptions;
 using System;
 using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Security.Claims;
+using System.Text;
 using System.Threading.Tasks;
 
 namespace SchedulearnBackend.Services
 {
     public class UserService
     {
+        private readonly IConfiguration _configuration;
         private readonly SchedulearnContext _schedulearnContext;
         private readonly TeamService _teamService;
         private readonly LimitService _limitService;
 
-        public UserService(TeamService teamService, LimitService limitService, SchedulearnContext schedulearnContext)
+        public UserService(IConfiguration configuration, TeamService teamService, LimitService limitService, SchedulearnContext schedulearnContext)
         {
+            _configuration = configuration;
             _schedulearnContext = schedulearnContext;
             _teamService = teamService;
             _limitService = limitService;
         }
 
+        public async Task<string> Authenticate(string userEmail, string userPassword)
+        {
+            var user = await _schedulearnContext.Users.Where(u => u.Email == userEmail && u.Password == userPassword).FirstOrDefaultAsync();
+            if (user != null)
+            {
+                var claims = new[] {
+                    new Claim(JwtRegisteredClaimNames.Sub, _configuration["Jwt:Subject"]),
+                    new Claim(JwtRegisteredClaimNames.Jti, Guid.NewGuid().ToString()),
+                    new Claim(JwtRegisteredClaimNames.Iat, DateTime.UtcNow.ToString()),
+                    new Claim("Id", user.Id.ToString()),
+                    new Claim("Name", user.Name),
+                    new Claim("Surname", user.Surname),
+                    new Claim("Email", user.Email)
+                   };
+                var key = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(_configuration["Jwt:Key"]));
+                var signIn = new SigningCredentials(key, SecurityAlgorithms.HmacSha256);
+                var token = new JwtSecurityToken(_configuration["Jwt:Issuer"], _configuration["Jwt:Audience"], claims, expires: DateTime.UtcNow.AddDays(1), signingCredentials: signIn);
+
+                return new JwtSecurityTokenHandler().WriteToken(token);
+            }
+
+            throw new UnauthorisedException($"Username or password is incorrect");
```

Comment:
Why not invert this logic and throw this when user is null, otherwise authenticate? Seems weird you check this at the very top of your code, but throw appropriate error at the end.